var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -468,7 +511,10 @@ class Mfem(Package, CudaPackage, ROCmPackage):
         when="@4.6.0 +gslib+shared+miniapps",
         sha256="2a31682d876626529e2778a216d403648b83b90997873659a505d982d0e65beb",
     )
-    patch("mfem-hip.patch", when="+rocm ^hip@6.0:")
+    # FIXME: if we need this patch for v4.7.0, we need a new patch since this
+    #        one fails
+    patch("mfem-hip.patch", when="@4.6.0 +rocm ^hip@6.0:")
```

Comment:
Hi @renjithravindrankannath, it looks like you added this patch in commit 118ff338a, so I wanted to understand the issue it fixes and figure out if we need a new patch for mfem v4.7.

It looks like with HIP >= 6, the patch and the other change in 118ff338a add explicit linking with `libamdhip64` which seems strange -- isn't that library automatically linked with when using `hipcc` (for compiling and/or linking) as does mfem?

Reply:
@nychiang can you please take a look?

Reply:
> Hi @renjithravindrankannath, it looks like you added this patch in commit [118ff33](https://github.com/spack/spack/commit/118ff338abbf242712b822f2bffa954eb46654e5), so I wanted to understand the issue it fixes and figure out if we need a new patch for mfem v4.7.
> 
> It looks like with HIP >= 6, the patch and the other change in [118ff33](https://github.com/spack/spack/commit/118ff338abbf242712b822f2bffa954eb46654e5) add explicit linking with `libamdhip64` which seems strange -- isn't that library automatically linked with when using `hipcc` (for compiling and/or linking) as does mfem?

Hi @v-dobrev ,
This patch was added when it was unable to find libhsa-runtime64.so and load libamdhip64.so. [PR 44171](https://github.com/spack/spack/pull/44171) fixed the issue and patch can be removed for 6.1.

Reply:
> @nychiang can you please take a look?

@cnpetra, @nychiang, just FYI: this particular discussion in the PR is not related to HiOp -- you probably just want to look at the changes in the HiOp Spack package.

Reply:
> > Hi @renjithravindrankannath, it looks like you added this patch in commit [118ff33](https://github.com/spack/spack/commit/118ff338abbf242712b822f2bffa954eb46654e5), so I wanted to understand the issue it fixes and figure out if we need a new patch for mfem v4.7.
> > It looks like with HIP >= 6, the patch and the other change in [118ff33](https://github.com/spack/spack/commit/118ff338abbf242712b822f2bffa954eb46654e5) add explicit linking with `libamdhip64` which seems strange -- isn't that library automatically linked with when using `hipcc` (for compiling and/or linking) as does mfem?
> 
> Hi @v-dobrev , This patch was added when it was unable to find libhsa-runtime64.so and load libamdhip64.so. [PR 42876](https://github.com/spack/spack/pull/42876) fixed the issue and patch can be removed for 6.1.

Thanks for the clarification, @renjithravindrankannath. If I understand correctly the Spack PR you linked (#42876) also fixes the issue for HIP v6.0 (by applying a patch present in HIP v6.1), so it seems in the latest Spack we don't need the patch you added to MFEM neither for HIP v6.0 nor for HIP v6.1. Do you agree? Or did I misunderstand?

Reply:
Looking again at #42876, it looks like the patch there is applied only when using Spack-built HIP v6.0. If using external HIP v6.0, the issue may still be present. I'll try to test with external HIP v6.0 and add an updated patch for mfem v4.7, if needed.

Reply:
> > > Hi @renjithravindrankannath, it looks like you added this patch in commit [118ff33](https://github.com/spack/spack/commit/118ff338abbf242712b822f2bffa954eb46654e5), so I wanted to understand the issue it fixes and figure out if we need a new patch for mfem v4.7.
> > > It looks like with HIP >= 6, the patch and the other change in [118ff33](https://github.com/spack/spack/commit/118ff338abbf242712b822f2bffa954eb46654e5) add explicit linking with `libamdhip64` which seems strange -- isn't that library automatically linked with when using `hipcc` (for compiling and/or linking) as does mfem?
> > 
> > 
> > Hi @v-dobrev , This patch was added when it was unable to find libhsa-runtime64.so and load libamdhip64.so. [PR 42876](https://github.com/spack/spack/pull/42876) fixed the issue and patch can be removed for 6.1.
> 
> Thanks for the clarification, @renjithravindrankannath. If I understand correctly the Spack PR you linked (#42876) also fixes the issue for HIP v6.0 (by applying a patch present in HIP v6.1), so it seems in the latest Spack we don't need the patch you added to MFEM neither for HIP v6.0 nor for HIP v6.1. Do you agree? Or did I misunderstand?

Hi @v-dobrev , Yes, You are right. Both 6.0 and 6.1 doesn't need the patch.

Reply:
I tested with external hip-6.0.0 and the without the `mfem-hip.patch` -- both mfem-4.6.0 and mfem-4.7.0 worked fine. Therefore, I decided to remove the patch altogether, see 05d4e606da.